### PR TITLE
java implementation

### PR DESCRIPTION
**Pull request by edelfinato**
A simple java implementation using SpringBoot and Jersey.

### CODE REVIEW

File: java/blockchain/src/test/java/org/learn/blockchains/BlockchainApplicationTests.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import org.junit.runner.RunWith;
+import org.learn.blockchains.components.BlockChain;
+import org.learn.blockchains.model.Block;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
```
Comment (psinghcredify) [inline java/blockchain/src/test/java/org/learn/blockchains/BlockchainApplicationTests.java]: the test cases  written don't need a server to be started.

Comment (pravsingh) [inline java/blockchain/src/test/java/org/learn/blockchains/BlockchainApplicationTests.java]: current test cases don't need server to be started, they are just doing component level testing.

**Comment by michaelwayman** [comment]
maybe I'm just not familiar enough with Java but there is a LOT of extra noise in here..
perhaps you could just stick with the source code?
the `.gitignore` alone seems daunting unmaintainable now..
I think you should be able to just have the source code and throw in a docker file or small README with instructions on how to run, I'm not really sure what any of that other stuff has to do with this project.

File: java/blockchain/src/test/java/org/learn/blockchains/BlockchainApplicationTests.java (hunk 1 above)
Comment (edelfinato) [reply java/blockchain/src/test/java/org/learn/blockchains/BlockchainApplicationTests.java]: I removed it. Thanks for the suggestion.

**Comment by edelfinato** [comment]
@michaelwayman I'll add a simple README. Thanks for the suggestion. 


> the .gitignore alone seems daunting unmaintainable now..

I'll put .gitingore only in java directory Thanks.

**Comment by pravsingh** [comment]
my version of the implementation is at https://github.com/pravsingh/Blockchain.
Few test cases are testing end to end flow and demonstrating on how it works.

**Comment by dvf** [comment]
Closing for now.

**Pull request closed without merging**